src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticHelper.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeLocalFunctionStatic
+{
+    internal static class MakeLocalFunctionStaticHelper
+    {
+        public static bool IsStaticLocalFunctionSupported(SyntaxTree tree)
+            => tree.Options is CSharpParseOptions csharpOption && csharpOption.LanguageVersion >= LanguageVersion.CSharp8;
+
+        public static bool TryGetCaputuredSymbols(LocalFunctionStatementSyntax localFunction, SemanticModel semanticModel, out ImmutableArray<ISymbol> captures)
+        {
+            var dataFlow = semanticModel.AnalyzeDataFlow(localFunction);
+            captures = dataFlow.CapturedInside;
+
+            return dataFlow.Succeeded;
+        }
+
+        public static async Task<Document> MakeLocalFunctionStaticAsync(
+            LocalFunctionStatementSyntax localFunction,
+            ImmutableArray<ISymbol> captures,
+            SemanticModel semanticModel,
+            Document document,
+            CancellationToken cancellationToken)
+        {
+            var root = (await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false))!;
```

Comment:
Another one where suppression operator is just a burden - can we add `GetNonNullSyntaxRootAsync` with appropriate doc comments?

Reply:
👍 on that.  `GetRequiredSyntaxRootAsync` would be my preference on naming.

Reply:
That name looks reasonable as well.

---
In reply to: [324847614](https://github.com/dotnet/roslyn/pull/37629#discussion_r324847614) [](ancestors = 324847614)

Reply:
Related to https://github.com/dotnet/roslyn/issues/38326

Reply:
I will probably create an analyzer to see what Roslyn APIs we apply `!` on, that would give us an idea.